dnn/data/converters/get_purdue.py

```diff
@@ -0,0 +1,97 @@
+import json
+import xml.dom.minidom as minidom
+
+import requests
+from absl import app, flags
+from absl.flags import FLAGS
+from tqdm import tqdm
+import os
+import time
+
+flags.DEFINE_string('input', '../Purdue Dataset/label.json',
+                    'the path for input ROCO dataset (please unzip by yourself)')
+flags.DEFINE_string('output', '../Purdue Dataset',
+                    'the path for input ROCO dataset (please unzip by yourself)')
+
+
+def main(_argv):
+    json_file = json.loads(open(FLAGS.input, 'rb').read())['_via_img_metadata']
+    if not os.path.exists(os.path.join(FLAGS.output, 'image')):
+        os.mkdir(os.path.join(FLAGS.output, 'image'))
+    if not os.path.exists(os.path.join(FLAGS.output, 'image_annotation')):
+        os.mkdir(os.path.join(FLAGS.output, 'image_annotation'))
+    for m, image_id in zip(tqdm(range(len(json_file)),
+                                desc='Getting Purdue Dataset: ',
+                                unit='pic',
+                                ncols=150),
+                           json_file):
+        if m < 0:  # to resume download process, set 0 to be your latest download picture index
+            continue
+        url = json_file[image_id]['filename']
+        try:
+            r = requests.get(url, timeout=10)
+        except requests.exceptions.RequestException:
+            time.sleep(10)
+            r = requests.get(url, timeout=30)
```

Comment:
Add some verbose logging here?

Reply:
This script would retry connection once, and the retry process should not appear in logging message (too noisy).